src/write_gcm.c

```diff
@@ -2680,9 +2696,12 @@ static int wg_write(const data_set_t *ds, const value_list_t *vl,
   pthread_mutex_lock(&queue->mutex);
   // Backpressure. If queue is backed up then something has gone horribly wrong.
   // Maybe the queue processor died.
-  if (queue->size > QUEUE_DROP_SIZE) {
-    INFO("write_gcm: Dropping data point because queue has size %zd",
-        queue->size);
+  if (queue->size >= QUEUE_DROP_SIZE) {
+    INFO("write_gcm: Clearing queue because it has size %zd", queue->size);
```

Comment:
This seems like the wrong behavior -- if we end up in a state where it is backed up, we'll just discard everything which could mean real data loss for customers.  I think instead we should be keeping a buffer up to some maximum size and then discarding old entries as new ones come in rather than just dropping the whole thing.  That's basically how things end up working today and seems pretty reasonable